sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/bridge/ValidatingStargateBridge.java

```diff
@@ -0,0 +1,171 @@
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/** A mock bridge implementation for unit tests. */
+public class ValidatingStargateBridge implements StargateBridge {
+
+  private final List<QueryExpectation> expectedQueries = new ArrayList<>();
+
+  public void reset() {
+    expectedQueries.clear();
+  }
+
+  public void validate() {
+    expectedQueries.forEach(QueryExpectation::validate);
+  }
+
+  @Override
+  public Uni<QueryOuterClass.Response> executeQuery(QueryOuterClass.Query query) {
+    QueryExpectation expectation =
+        findQueryExpectation(query.getCql(), query.getValues().getValuesList());
+    return expectation.execute(null);
+  }
+
+  @Override
+  public Uni<QueryOuterClass.Response> executeBatch(QueryOuterClass.Batch batch) {
+    return batch.getQueriesList().stream()
+        .map(
+            query -> {
+              QueryExpectation expectation =
+                  findQueryExpectation(query.getCql(), query.getValues().getValuesList());
+              return expectation.execute(batch.getType());
+            })
+        // Return the last result
+        .reduce((first, second) -> second)
+        .orElseThrow(() -> new AssertionError("Batch should have at least one query"));
+  }
+
+  private QueryExpectation findQueryExpectation(String cql, List<QueryOuterClass.Value> values) {
+    return expectedQueries.stream()
+        .filter(q -> q.matches(cql, values))
+        .findFirst()
+        .orElseThrow(
+            () ->
+                new AssertionError(
+                    String.format(
+                        "Unexpected query, should have been mocked with withQuery(): %s", cql)));
+  }
+
+  @Override
+  public Uni<Schema.CqlKeyspaceDescribe> describeKeyspace(Schema.DescribeKeyspaceQuery request) {
+    throw new UnsupportedOperationException("Not implemented by this mock");
+  }
+
+  @Override
+  public Uni<Schema.AuthorizeSchemaReadsResponse> authorizeSchemaReads(
+      Schema.AuthorizeSchemaReadsRequest request) {
+    throw new UnsupportedOperationException("Not implemented by this mock");
+  }
+
+  @Override
+  public Uni<Schema.SupportedFeaturesResponse> getSupportedFeatures(
+      Schema.SupportedFeaturesRequest request) {
+    throw new UnsupportedOperationException("Not implemented by this mock");
+  }
+
+  private QueryExpectation add(QueryExpectation expectation) {
+    expectedQueries.add(expectation);
+    return expectation;
+  }
+
+  public QueryExpectation withQuery(String cql, QueryOuterClass.Value... values) {
+    return add(new QueryExpectation(Pattern.quote(cql), Arrays.asList(values)));
+  }
+
+  public abstract static class QueryAssert {
+
+    private final AtomicInteger executeCount = new AtomicInteger();
+
+    public AbstractIntegerAssert<?> assertExecuteCount() {
+      return assertThat(executeCount.get());
+    }
+
+    void executed() {
+      executeCount.incrementAndGet();
+    }
+  }
+
+  public static class QueryExpectation extends QueryAssert {
```

Comment:
I know it's needed atm, but as you are on it, can we add a support for validation if the query was `enriched`. Needed for the reading side..